### PR TITLE
[s] Hallucinations 4.0

### PR DESCRIPTION
**Pull request by XDTM**
No ingame changelog cause these are supposed to fool people.

Changes:
- Hallucinations now pick from a single weighted list, instead of being separated in tiers. The more obvious a hallucinations, the less likely it is to appear.
- Hallucination is now a contantly ticking down variable. 1 hallucination = 2 seconds duration. Hallucinating effects have been rebalanced to fit this.
- Hallucinations have a random cooldown ranging from 10 to 60 seconds. The status will still tick down in the meantime.
- The Sounds hallucinations has been split in three: realistic sounds (likely), weird sounds (rare), and station messages (i.e. the shuttle arrived, blob warning, etc.)(also rare).
- The Bolts hallucination now stops you from bumping doors open. You can still click them to open them.
- Removed the singulo hallucination. It's pretty much impossible to make it realistic enough, and it's too widely known to scare someone. It's an annoyance _and_ a definite tell that you're hallucinating.
- Removed the xeno attack for the same reasons. It was kept in the code to serve as a pseudotemplate for hallucination mobs.
- Removed the fake attacker hallucination. Trying to replicate full attack and defense code as well as managing icon generation with blacklists and all is just not worth it. It would never be realistic enough.
- RDS hallucinations have been either rolled into normal ones or removed (the OOC ones). No reason to keep them separate, aside from throwing off the hallucination odds (which has resulted in complaints about the cult victory warning being too common).
- Added/edited a lot of strings for the text-based hallucinations. I could definitely use a few more for the message hallucination, i got writer's block there.

### CODE REVIEW

**Comment by demonfiren** [comment]
>removed fake attacker
than fuck that was never not annoying as fuck

**Comment by Mickyan** [comment]
I could never stand the ooc hallucinations in RDS so I might actually use it now

You know perhaps for mob hallucinations it would be interesting if they could be controlled by ghosted players, assuming that's even possible

**Comment by Barhandar** [comment]
Do they _need_ to be realistic?

**Comment by XDTM** [comment]
@Barhandar hallucinations will never be believed 100% of the time, but my goal is to increase the amount of times it fools people into acting in interesting ways, while reducing the amount of times hallucinations annoy people.

More "oh shit i killed the AI because someone on the radio saw a blue APC and it was actually a hallucination", less "i just got randomly stunned".

**Comment by arsserpentarium** [comment]
I think, that hallucinations may be different from different drugs/causes.
So don't delete singulo/fake attacker. And leave it for case of extreme hallucinations. Like being too high on spess drugs.

**Comment by Shadowflame909** [comment]
You fool. If you make hallucinations realistic you're gonna trick people into getting banned for killing the ai over a hallucination. "Why did you do that." "I saw this incredibly realistic hallucination where the apc turned blue. The doors bolted and shocked me." "Being insane is never an argument." You have been permanently banned. Talk to the coders to Nerf hallucinations.

**Comment by GuppyLaxx** [comment]
I liked the OOC hallucinations. In the sense of the concept, how they were put in was atrocious. If you want some ideas for that, fake bwoinks and the like. If you feel like adding another rare hallucination, shuttle being called with some small random reasons like "shits fucked" would work. Also fucking, there's a hacking hallucination sound that's `really` obvious due to the use of the old hacking sounds.

I'm still not 100% straight on how this is gonna affect RDS, but out of that it seems to be fine, aslong as I'm not having my chat spammed by NAR SIE HAS RISEN, constantly being on fire, and getting shocked, it's good

**Comment by XDTM** [comment]
@GuppyLaxx OOC is not ok because stuff like an admin contacting you should not be confused for an ingame mechanic, ever.

RDS will now just be intermittent hallucinations, and those should be less frequent and obnoxious.

I thought about the shuttle call hallucination, but unless i add a really wide list of motivations it's gonna be easy to detect as a hallucination.

About the hacking hallucination, i removed the clicking sounds inbetween. Maybe i should play the game once in a while.

**Comment by XDTM** [comment]
@Shadowflame909 jokes aside there is fairly precise logging for hallucinations, so you can verify is the player has been fooled or is trying to fool you

**Comment by Shadowflame909** [comment]
Oh, nevermind then. Let the hallucinations continue to get realistic!

**Comment by newfren** [comment]
We definitely have directional sound, yes.

**Comment by XDTM** [comment]
Already did! I forgot to put that in the change list.

**Comment by XDTM** [comment]
In this commit:
- Sounds now have a localized, random source instead of being centered on the victim. Definitely more realistic.
- The Items-Other hallucination now properly picks a random tartget instead of the first available.
- Made a Shock hallucination. Now the door shock uses the hallucination instead of local code. Fixed the black outline not falling down with you.
- Reworked the dangerflash hallucination; now creates interactive hallucination effects. Currently: Lava (sets you of fake fire), Chasm ("drops" you), Electric Anomaly (shocks you). Open to more ideas.
- Whispers' speech is now always capitalized.
-

**Comment by XDTM** [comment]
I want to replace husks with a hallucination that shows creepy or weird stuff/shapes in the distance. If anyone has some sprites that could fit that purpose i'd love to add them

**Comment by Xhuis** [comment]
not a fan of the rds removal, the point of the hallucinations was to make the trait unique in a way that wasn't just "occasional hallucinations" and why the trait was made so wordy and unique to begin with. if you want to remove the only thing that makes it unique, rename the trait.

**Comment by XDTM** [comment]
The hallucinations it had were of the same category of existing hallucinations (i.e narsie rising is pretty much the same as the AI delta alert). Those that weren't were instead OOC-based (i.e. the big messages saying to leave the server or close the game window) which, last i've heard, are a no-no, aside from being really obvious after the first time.

I would've kept special hallucinations if they were actually special and not just kept apart for the sake of it.

**Comment by Xhuis** [comment]
The 4th-wall breaking traits have never been brought up as problematic until this PR when you cut them. There's ultimately nothing I can do to keep it as-is, but part of the reason why RDS is named so uniquely and has such a wordy description and disclaimer is because you see and hear things with RDS that you don't see with anything else. I'll insist again that if you remove the unique traits that you should make it more subtle in naming or description.

**Comment by ExcessiveUseOfCobblestone** [comment]
Regardless of if it's never been brought up or not, having immersion breaking hallucinations is not cool outside of ooc situations, which would also probably not be cool then either.

File: code/modules/flufftext/Hallucination.dm [hunk 1]
```diff
@@ -325,151 +305,195 @@ GLOBAL_LIST_INIT(hallucinations_major, list(
...
 /datum/hallucination/battle
-	cost = 15
 
-/datum/hallucination/battle/New(mob/living/carbon/T, forced = TRUE, battle_type)
+/datum/hallucination/battle/New(mob/living/carbon/C, forced = TRUE, battle_type)
 	set waitfor = FALSE
 	..()
-	var/hits = rand(3,6)
+	var/list/turf/sourcelocs = list()
+	for(var/turf/open/T in range(world.view+1,target)-range(world.view-2,target))
```
Comment (AnturK) [inline code/modules/flufftext/Hallucination.dm]: If you're just picking one random offset is better than iterating all these (twice). Could probably pick from off-view locations only alternatively.

**Comment by improvedname** [comment]
New brain tumor fake attackee you get a fake that attacts you 24/7

**Comment by improvedname** [comment]
Also keep shuttle hallicination which makes a fake shuttle appear so you walk into it but instead your spacing yourself

**Comment by XDTM** [comment]
@AnturK should be fine now

**Pull request merged**